Update to v0.1.1

kenju opened the pull request:
- Update gem version to v0.1.1
- Add CHANGELOG.md
- `ts_assets.gemspec` のtypo修正など

@bitjourney/kibela-developers レビューお願いします 🙏

kenju committed:
add CHANGELOG.md

kenju committed:
remote gem push restrictions

kenju committed:
add version specifications in gemspec

kenju committed:
update version to v0.1.1

gfx commented on CHANGELOG.md:
```diff
@@ -0,0 +1,6 @@
+
+## v0.1.1 - 2017/7/6
+
+* Add `TsAssets.generate` to generate react-components in TypeScript
```
`initial version` で十分です。

kenju replied:
`initial version` になおしております！

https://github.com/bitjourney/ts_assets-rails/pull/2/commits/3f262e36ef8542184f7fb007d3dc97ceb225c91d

michiomochi commented on ts_assets.gemspec:
```diff
@@ -31,10 +22,10 @@ Gem::Specification.new do |spec|
+  spec.add_runtime_dependency 'sprockets', "~> 3.7"
 
-  spec.add_development_dependency 'pry'
+  spec.add_development_dependency 'pry', "~> 0.10"
```
バージョン指定は全てしたほうがいいんですかね？
最低限でいいきがします。
つまりpryとかは縛るひつようないかなと。

gfx replied:
バージョン指定はしてもいいですが（Rakeとかはわりと非互換がある）、 `>= x` にしましょう。

kenju replied:
開発環境は割とゆるくてもいいかなと思ったので、`runtime_dependency` の方だけ `>= x`に変更しました。

https://github.com/bitjourney/ts_assets-rails/pull/2/commits/4c11071c575d40f2f64db463a609f03a871aa540

`gem build ts_assets.gemspec` コマンドを実行すると下記警告が出たのでつけましたが、公開するgemは少なくともバージョン指定をするのがお作法的なもの、とかでもないんですかね 🤔 

```
WARNING:  open-ended dependency on pry (>= 0, development) is not recommended
  if pry is semantically versioned, use:
    add_development_dependency 'pry', '~> 0'
```

gfx replied:
runtime depsも `~>` ではなく `>=` で指定すべきです。  `~>` はよほど特別な理由でもないかぎり避けてください。

理由: https://www.hsbt.org/diary/20160829.html

kenju replied:
なるほど、特に理由もなくバージョン指定すると↓のような事態になるリスクがあるんですね

> もしくは特に意味合いはないものの保守的に '~> 1.8' を指定しているものが多いものの、以下のいずれかの対応を行わなくては Ruby 2.4.0 以降に生き残ることはできないことが現時点で概ね確定している。

対応しました https://github.com/bitjourney/ts_assets-rails/pull/2/commits/ccf934edc95f56bb78ac0ea9e9ad387d8add4f4c

kenju committed:
change semver for dependencies from '~>' to '>='

because it increases risk for breaking changes
on upgrading some libs/frameworks or ruby itself

why) https://www.hsbt.org/diary/20160829.html

michiomochi commented:
LGTM！ 👍

gfx commented:
LGTM
